.github/workflows/workflow.yaml

```diff
@@ -59,7 +59,7 @@ jobs:
           pytest -v --run-cfchecks
   docs:
     name: Build and deploy the documentation
-    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags') && contains(github.ref, 'refs/tags/0.')
```

Comment:
Actually I think it may be easiest to check the tags within the `on:` block in some way? e.g. https://stackoverflow.com/questions/58862864/github-actions-ci-conditional-regex

Reply:
We need to make sure that actions run on any tag, so it is kept as `*`, the rest of the build doc and build+publishing needs on release tags only hence did not directly update the on parameter in yaml file.

Reply:
Ok, I feel like this could be a benefit of splitting this out more into separate files but, having said that, if we need to run tests etc. before building and publishing anyway that may not be particularly easy and if statements may be the only way to go.

Regardless, I think we do want to try and add a bit more flexibility that "0." versions as we run the risk of problems down the line if we/when change our major version.

Reply:
This approach is done in cases of pre-release tags e.g in our case it can be `all-parsers' written as tag then it will run the other necessary jobs but if tag is 0. then only build doc, build+publish package will run. Having `on` with specific regex means the action will only run for those types of tags

Reply:
If we want to cover this case then I can write to handle any number